src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs

```diff
@@ -172,13 +172,29 @@ public void Listen(int maxIncoming = 20)
             {
                 while (!this.serverCancel.IsCancellationRequested)
                 {
+                    // Used to record any errors occurrring in the thread pool task.
+                    Exception err = null;
+
                     TcpClient tcpClient = await Task.Run(() =>
                     {
-                        Task<TcpClient> acceptTask = this.tcpListener.AcceptTcpClientAsync();
-                        acceptTask.Wait(this.serverCancel.Token);
-                        return acceptTask.Result;
+                        try
+                        {
+                            Task<TcpClient> acceptTask = this.tcpListener.AcceptTcpClientAsync();
+                            acceptTask.Wait(this.serverCancel.Token);
+                            return acceptTask.Result;
+                        }
+                        catch (Exception e)
+                        {
+                            // Record the error.
+                            err = e;
+                            return null;                            
+                        }
                     }).ConfigureAwait(false);
 
+                    // Raise the error.
+                    if (err != null)
```

Comment:
pls be consistent with the previous code, either throw in both on new line or same line

Reply:
Done